pandas/tests/test_errors.py

```diff
@@ -110,3 +110,8 @@ def test_AbstractMethodError_classmethod():
     xpr = "This method must be defined in the concrete class Foo"
     with pytest.raises(AbstractMethodError, match=xpr):
         Foo().method()
+
+    types = {"method", "classmethod", "staticmethod", "property"}
+    xpr = f"methodtype must be one of {types}, got unknownmethod instead."
```

Comment:
let's avoid logic in tests if we can - can you write the raw string please?

ideally, it would probably be good to have an error message like
```
'methodtype' must be one of {'method', 'classmethod', 'staticmethod', 'property'}, got 'unknownmethod' instead
```

Reply:
https://github.com/pandas-dev/pandas/pull/51484/commits/b697aba60edcfade44c09e904e85c7aa5663711b